### PR TITLE
Few fixes for editor tab in editor settings

### PR DESCRIPTION
**Pull request by Chaosus**
While I agree that **main_font_size** require restart to set full effect on UI, the other font options are not have any visible changes before and after restart, and apply immediatly and correctly. Also removes **show_type_icons** flag cuz its not have any effect in new inspector property editor.

### CODE REVIEW

File: editor/property_editor.cpp [hunk 1]
```diff
@@ -4394,7 +4394,7 @@ PropertyEditor::PropertyEditor() {
 	use_filter = false;
 	subsection_selectable = false;
 	property_selectable = false;
-	show_type_icons = EDITOR_DEF("interface/editor/show_type_icons", false);
+	show_type_icons = false;
```
Comment (akien-mga) [inline editor/property_editor.cpp]: At the very least there should be a comment explaining that it's not working and has to be ported to the new system. If we're just removing the option, we might just as well remove every reference to `show_type_icons`, as is it's dead code.

**Comment by akien-mga** [comment]
> While I agree that main_font_size require restart to set full effect on UI, the other font options are not have any visible changes before and after restart, and apply immediatly and correctly.

What about `main_font` though? Different fonts have different sizes for a given `main_font_size`, so they would impact the UI too I suppose?

**Comment by Chaosus** [comment]
> What about main_font though?
 I detect that if you change it, the labels are not have the same positions than after restart.

File: editor/property_editor.cpp (hunk 1 above)
Comment (Chaosus) [reply editor/property_editor.cpp]: So, should I remove it or leave a comment ?

**Comment by Chaosus** [comment]
@akien-mga Alright, I removed old code related to **show_type_icons**

File: editor/property_editor.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/property_editor.cpp]: I guess leave a comment, this was a relatively popular feature (#15505) so I guess someone might want to fix their implementation in the new property editor.

Comment (Chaosus) [reply editor/property_editor.cpp]: Ok, I leave a comment

**Comment by Chaosus** [comment]
@akien-mga Ok, I've added a comment, as you requested

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**